_source/logzio_collections/_tracing-sources/nestjs-otel-auto.md

```diff
@@ -0,0 +1,488 @@
+* Running your NestJS application in conjunction with the OpenTelemetry instrumentation
+
+On deployment, the NestJS instrumentation automatically captures spans from your application and forwards them to the collector, which exports the data to your Logz.io account.
+
+</div>
+<!-- tab:end -->
+
+
+<!-- tab:start -->
+<div id="local-host">
+
+
+### Setup auto-instrumentation for your locally hosted NestJS application and send traces to Logz.io
+
+**Before you begin, you'll need**:
+
+* A NestJS application without instrumentation
+* An active account with Logz.io
+* Port `4317` available on your host system
+* A name defined for your tracing service
+
+
+<div class="tasklist">
+
+
+##### Download instrumentation packages
+
+Run the following command from the application directory:
+
+```shell
+npm install --save @opentelemetry/api
+npm install --save @opentelemetry/instrumentation
+npm install --save @opentelemetry/tracing
+npm install --save @opentelemetry/exporter-trace-otlp-http
+npm install --save @opentelemetry/resources
+npm install --save @opentelemetry/semantic-conventions
+npm install --save @opentelemetry/auto-instrumentations-node
+npm install --save @opentelemetry/sdk-node
+```
+
+##### Create a tracer file
+
+In the directory of your application file, create a file named `tracer.ts` with the following configuration:
+
+```javascript
+"use strict";
+
+const {
+    BasicTracerProvider,
+    ConsoleSpanExporter,
+    SimpleSpanProcessor,
+} = require("@opentelemetry/tracing");
+const { OTLPTraceExporter } = require("@opentelemetry/exporter-trace-otlp-http");
+const { Resource } = require("@opentelemetry/resources");
+const {
+    SemanticResourceAttributes,
+} = require("@opentelemetry/semantic-conventions");
+
+const opentelemetry = require("@opentelemetry/sdk-node");
+const {
+    getNodeAutoInstrumentations,
+} = require("@opentelemetry/auto-instrumentations-node");
+
+
+const exporter = new OTLPTraceExporter({
+    url: "http://localhost:4318/v1/traces"
+});
+
+const provider = new BasicTracerProvider({
+    resource: new Resource({
+        [SemanticResourceAttributes.SERVICE_NAME]:
+            "YOUR-SERVICE-NAME",
+    }),
+});
+// export spans to console (useful for debugging)
+provider.addSpanProcessor(new SimpleSpanProcessor(new ConsoleSpanExporter()));
+// export spans to opentelemetry collector
+provider.addSpanProcessor(new SimpleSpanProcessor(exporter));
+
+provider.register();
+const sdk = new opentelemetry.NodeSDK({
+    traceExporter: exporter,
+    instrumentations: [getNodeAutoInstrumentations()],
+});
+
+sdk
+    .start()
+    .then(() => {
+        console.log("Tracing initialized");
+    })
+    .catch((error) => console.log("Error initializing tracing", error));
+
+process.on("SIGTERM", () => {
+    sdk
+        .shutdown()
+        .then(() => console.log("Tracing terminated"))
+        .catch((error) => console.log("Error terminating tracing", error))
+        .finally(() => process.exit(0));
+});
+	
+```
+
```

Comment:
Can you add a line on YOUR_SERVICE_NAME? something like:
Replace <YOUR-SERVICE-NAME> with the name of your service